nuxeo-core/nuxeo-core-storage-sql/nuxeo-core-storage-sql/src/main/java/org/nuxeo/ecm/core/storage/sql/coremodel/SQLDocumentLive.java

```diff
@@ -339,12 +341,27 @@ public String getChangeToken() {
     }
 
     @Override
-    public boolean validateChangeToken(String changeToken) {
-        if (changeToken == null) {
+    public boolean validateChangeToken(String userChangeToken) {
+        if (userChangeToken == null) {
             return true;
         }
-        String currentToken = getChangeToken();
-        return validateChangeToken(changeToken, currentToken);
+        if (session.isChangeTokenEnabled()) {
+            Long sysVersion = (Long) getPropertyValue(Model.MAIN_SYS_VERSION_PROP);
+            Long changeToken = (Long) getPropertyValue(Model.MAIN_CHANGE_TOKEN_PROP);
+            return validateUserChangeToken(sysVersion, changeToken, userChangeToken);
+        } else {
+            Calendar modified;
+            try {
+                modified = (Calendar) getPropertyValue(DC_MODIFIED);
+            } catch (PropertyNotFoundException e) {
```

Comment:
Yes, for instance on [`DocumentModelImpl`](https://github.com/nuxeo/nuxeo/blob/release-9.1/nuxeo-core/nuxeo-core-api/src/main/java/org/nuxeo/ecm/core/api/impl/DocumentModelImpl.java#L1490)